master/internal/rm/kubernetesrm/pods.go

```diff
@@ -939,8 +947,27 @@ func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
 
 // summarize describes pods' available resources. When there's exactly one resource pool, it uses
 // the whole cluster's info. Otherwise, it matches nodes to resource pools using taints and
-// tolerations to derive that info.
-func (p *pods) summarize(ctx *actor.Context) (map[string]model.AgentSummary, error) {
+// tolerations to derive that info. This may be cached, so don't use this for decisions
+// that require up-to-date information.
+func (p *pods) summarize(ctx *actor.Context) (summary map[string]model.AgentSummary, err error) {
+	p.summarizeCacheLock.Lock()
+	defer p.summarizeCacheLock.Unlock()
+
+	if time.Since(p.summarizeCacheTime) < 5*time.Second {
+		return p.summarizeCache.summary, p.summarizeCache.err
+	}
+
+	// Keep the cache up to date. Note: since this only gets deferred when the cache is expired,
+	// we won't update the time unless we're generating a new value.
+	defer func() {
```

Comment:
This structure seems a little bit too clever. How about moving the real summarization into a `computeSummary` or something and then having this be `if time > 5 seconds { p.summarizeCache = computeSummary() } return p.summarizeCache...`?

Reply:
Good idea; I think that'd be way more readable.